src/plugins/kibana/server/routes/api/ingest/register_post.js

```diff
@@ -0,0 +1,101 @@
+const Boom = require('boom');
+const _ = require('lodash');
+const {templateToPattern, patternToTemplate} = require('../../../lib/convert_pattern_and_template_name');
+const indexPatternSchema = require('../../../lib/schemas/resources/index_pattern_schema');
+const handleESError = require('../../../lib/handle_es_error');
+const { convertToCamelCase } = require('../../../lib/case_conversion');
+const createMappingsFromPatternFields = require('../../../lib/create_mappings_from_pattern_fields');
+const initDefaultFieldProps = require('../../../lib/init_default_field_props');
+
+module.exports = function registerPost(server) {
+  server.route({
+    path: '/api/kibana/ingest',
+    method: 'POST',
+    config: {
+      validate: {
+        payload: indexPatternSchema
+      }
+    },
+    handler: function (req, reply) {
+      const callWithRequest = server.plugins.elasticsearch.callWithRequest;
+      const requestDocument = _.cloneDeep(req.payload);
+      const indexPatternId = requestDocument.id;
+      const indexPattern = convertToCamelCase(requestDocument);
+      delete indexPattern.id;
+
+      const mappings = createMappingsFromPatternFields(indexPattern.fields);
+      indexPattern.fields = initDefaultFieldProps(indexPattern.fields);
+
+      indexPattern.fields = JSON.stringify(indexPattern.fields);
+      indexPattern.fieldFormatMap = JSON.stringify(indexPattern.fieldFormatMap);
+
+      const patternCreateParams = {
+        index: '.kibana',
+        type: 'index-pattern',
+        id: indexPatternId,
+        body: indexPattern
+      };
+
+      callWithRequest(req, 'create', patternCreateParams)
+      .then((patternResponse) => {
+        return callWithRequest(req, 'indices.exists', {index: indexPatternId})
+        .then((matchingIndices) => {
+          if (matchingIndices) {
+            throw Boom.conflict('Cannot create an index pattern via this API if existing indices already match the pattern');
```

Comment:
Great catch. This was left over from when the API allowed patterns to be created for existing indices, but not templates. Updated it to do the check before creating anything.